modules/huddledmassesBidAdapter.js

```diff
@@ -0,0 +1,156 @@
+  '320x80': 59,
+  '640x480': 65,
+  '320x480': 67,
+  '320x320': 72,
+  '320x160': 73,
+  '480x300': 83,
+  '970x310': 94,
+  '970x210': 96,
+  '480x320': 101,
+  '768x1024': 102,
+  '1000x300': 113,
+  '320x100': 117,
+  '800x250': 118,
+  '200x600': 119
+};
+const INDEX_SIZE = {
+  1: '468x60',
+  2: '728x90',
+  10: '300x600',
+  15: '300x250',
+  19: '300x100',
+  43: '320x50',
+  44: '300x50',
+  48: '300x300',
+  54: '300x1050',
+  55: '970x90',
+  57: '970x250',
+  58: '1000x90',
+  59: '320x80',
+  65: '640x480',
+  67: '320x480',
+  72: '320x320',
+  73: '320x160',
+  83: '480x300',
+  94: '970x310',
+  96: '970x210',
+  101: '480x320',
+  102: '768x1024',
+  113: '1000x300',
+  117: '320x100',
+  118: '800x250',
+  119: '200x600'
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {object} bid The bid to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: (bid) => {
+    return (!isNaN(bid.params.placement_id) &&
+    ((bid.params.sizes != undefined && bid.params.sizes.length > 0 && bid.params.sizes.some((sizeIndex) => INDEX_SIZE[sizeIndex] != undefined)) ||
+    (bid.sizes != undefined && bid.sizes.length > 0 && bid.sizes.map((size) => `${size[0]}x${size[1]}`).some((size) => SIZE_INDEX[size] != undefined))));
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} validBidRequests A non-empty list of valid bid requests that should be sent to the Server.
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: (validBidRequests) => {
+    let winTop = window;
+    try {
+      window.top.location.toString();
+      winTop = window.top;
+    } catch (e) {
+      console.log(e);
+    };
+    let location = winTop.location;
+    let placements = [];
+    let request = {
+      'deviceWidth': winTop.screen.width,
+      'deviceHeight': winTop.screen.height,
+      'language': navigator ? navigator.language : '',
+      'secure': location.protocol === 'https:' ? 1 : 0,
+      'host': location.host,
+      'page': location.pathname,
+      'placements': placements
+    };
+    for (let i = 0; i < validBidRequests.length; i++) {
+      let bid = validBidRequests[i];
+      let placement = {};
+      placement['placementId'] = bid.params.placement_id;
+      placement['bidId'] = bid.bidId;
+      placement['sizes'] = bid.sizes;
+      placements.push(placement);
+    }
+    return {
+      method: 'POST',
+      url: URL,
+      data: request
+    };
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: (serverResponse) => {
+    let response = [];
+    try {
+      serverResponse = serverResponse.body;
```

Comment:
this line needs to be `serverResponse.body.body;`. Otherwise no bids are returned.

Reply:
Here I do not agree.
Example response:
``````
{
    "body": [{
        "width": 300,
        "height": 250,
        "ad": "<img src='http://supply.huddledmasses.com/img/logo.png' style='position: absolute; left: 50%; top:50%; margin-left: -113px; margin-top:-79px;'>",
        "requestId": "26042378f97b21",
        "cpm": 5,
        "ttl": 120,
        "creativeId": "123",
        "netRevenue": true,
        "currency": "USD",
        "dealId": "HASH"
    }]
}
``````

Reply:
it's broke. Run the code

Reply:
I cant understand whats wrong. 
Why doesn't it work for you.
Here is the test page with the given adapter:
http://hb-test.huddledmassessupply.com/
I do not understand why it does not work for you.

Reply:
@HuddledMasses 
The API changed. Please rebase to master. 

On your test page replace http://hb-test.huddledmassessupply.com/prebid-core.js with the latest code and you will see the issue.

Reply:
Here is the code changed: https://github.com/prebid/Prebid.js/blob/master/src/adapters/bidderFactory.js#L285

Reply:
I replaced the response at the server level.
That would not write in the code a long expression serverResponse.body.body. 
```````````
[{
        "width": 300,
        "height": 250,
        "ad": "<img src='http://supply.huddledmasses.com/img/logo.png' style='position: absolute; left: 50%; top:50%; margin-left: -113px; margin-top:-79px;'>",
        "requestId": "26042378f97b21",
        "cpm": 5,
        "ttl": 120,
        "creativeId": "123",
        "netRevenue": true,
        "currency": "USD",
        "dealId": "HASH"
}]
```````````